lighthouse-core/gather/driver.js

```diff
@@ -849,6 +849,24 @@ class Driver {
     });
   }
 
+  /**
+   * @param {number} [timeout]
+   * @return {Promise<LH.Crdp.Security.SecurityStateChangedEvent>}
+   */
+  getSecurityState(timeout = 1000) {
+    return new Promise((resolve, reject) => {
+      const err = new LHError(LHError.errors.SECURITY_STATE_TIMEOUT);
+      const asyncTimeout = setTimeout((_ => reject(err)), timeout);
+
+      this.sendCommand('Security.enable');
+      this.once('Security.securityStateChanged', (e) => {
```

Comment:
somewhat tangential... 

@paulirish are we just paranoid for doing all our listeners before enabling or do we actually need to? I seem to remember dgozman scolding me for not trusting in JS microtasks when we had listeners before `.enable` :)

Reply:
hmm good call.

This code here would be a problem if there was an `await` on the `.enable` call above. which seems fairly reasonable.

So yeah +1 to defining the once handler before we flip enable() on

Reply:
good point, fixed.

I don't quite understand the execution model here. It's clear to me why `await`ing the enable would schedule registering the listener much too late, but I don't understand how this original code worked (enable without `await`ing, then register a listener via `.once`). Why does enabling the Security domain still occur after the listener is registered?